modules/apps/portal-workflow/portal-workflow-web/src/main/java/com/liferay/portal/workflow/web/internal/groovy/script/uses/factory/WorkflowDefinitionGroovyScriptUsesFactory.java

```diff
@@ -0,0 +1,148 @@
+	implements GroovyScriptUsesFactory {
+
+	@Override
+	public List<GroovyScriptUse> create(ResourceRequest resourceRequest)
+		throws Exception {
+
+		return TransformUtil.transform(
+			_workflowDefinitionManager.getActiveWorkflowDefinitions(
+				QueryUtil.ALL_POS, QueryUtil.ALL_POS),
+			workflowDefinition -> {
+				String content = workflowDefinition.getContentAsXML();
+
+				if (!content.contains(
+						"<script-language>groovy</script-language>")) {
+
+					return null;
+				}
+
+				Company company = _companyLocalService.getCompany(
+					workflowDefinition.getCompanyId());
+
+				Locale locale = resourceRequest.getLocale();
+
+				return new GroovyScriptUse(
+					company.getWebId(),
+					workflowDefinition.getTitle(locale.getLanguage()),
+					_getSourceURL(company, workflowDefinition));
+			});
+	}
+
+	private String _getBaseURL(Company company, String portletId)
+		throws Exception {
+
+		String url = StringBundler.concat(
+			company.getPortalURL(GroupConstants.DEFAULT_PARENT_GROUP_ID),
+			PropsValues.LAYOUT_FRIENDLY_URL_PRIVATE_GROUP_SERVLET_MAPPING,
+			GroupConstants.CONTROL_PANEL_FRIENDLY_URL,
+			PropsValues.CONTROL_PANEL_LAYOUT_FRIENDLY_URL);
+
+		url = HttpComponentsUtil.addParameter(url, "p_p_id", portletId);
+		url = HttpComponentsUtil.addParameter(url, "p_p_lifecycle", "0");
+		url = HttpComponentsUtil.addParameter(
+			url, "p_p_state", WindowState.MAXIMIZED.toString());
+
+		return HttpComponentsUtil.addParameter(
+			url, "p_p_mode", PortletMode.VIEW.toString());
+	}
+
+	private String _getSourceURL(
+			Company company, WorkflowDefinition workflowDefinition)
+		throws Exception {
+
+		if (_workflowPortletTabServiceTracker.contains(
+				WorkflowWebKeys.WORKFLOW_TAB_DEFINITION)) {
+
+			String url = _getBaseURL(
+				company, WorkflowPortletKeys.KALEO_DESIGNER);
+
+			String namespace = _portal.getPortletNamespace(
+				WorkflowPortletKeys.KALEO_DESIGNER);
+
+			url = HttpComponentsUtil.addParameter(
+				url, namespace + "clearSessionMessage", true);
+			url = HttpComponentsUtil.addParameter(
+				url, namespace + "draftVersion",
+				workflowDefinition.getVersion() + ".0");
+			url = HttpComponentsUtil.addParameter(
+				url, namespace + "mvcPath",
+				"/designer/edit_workflow_definition.jsp");
+			url = HttpComponentsUtil.addParameter(
+				url, namespace + "name", workflowDefinition.getName());
+
+			return HttpComponentsUtil.addParameter(
+				url, namespace + "redirect",
+				_getBaseURL(
+					company, WorkflowPortletKeys.CONTROL_PANEL_WORKFLOW));
+		}
+
+		String url = _getBaseURL(
+			company, WorkflowPortletKeys.CONTROL_PANEL_WORKFLOW);
+
+		String namespace = _portal.getPortletNamespace(
+			WorkflowPortletKeys.CONTROL_PANEL_WORKFLOW);
+
+		url = HttpComponentsUtil.addParameter(
+			url, namespace + "mvcPath",
+			"/definition/edit_workflow_definition.jsp");
+
+		return HttpComponentsUtil.addParameter(
+			url, namespace + "redirect",
+			HttpComponentsUtil.addParameter(
+				_getBaseURL(
```

Comment:
You are invoking `_getBaseURL(company, WorkflowPortletKeys.CONTROL_PANEL_WORKFLOW)` twice here, maybe you can reuse the value from the first invoke to avoid more than one call.

Reply:
Done [here](https://github.com/liferay-workflow/liferay-portal/pull/1103/commits/a6196db56ed019ce0b49d8c6d06d90a932f51bbd)